pkgs/development/libraries/openmpi/default.nix

```diff
@@ -51,19 +64,28 @@ stdenv.mkDerivation rec {
     ++ lib.optionals cudaSupport [ cudaPackages.cuda_nvcc ]
     ++ lib.optionals fortranSupport [ gfortran ];
 
-  configureFlags = lib.optional (!cudaSupport) "--disable-mca-dso"
+  configureFlags =
+    [
+      # UCX support is recommended to use with cuda for the most robust OpenMPI build
+      # https://github.com/openucx/ucx
+      # https://www.open-mpi.org/faq/?category=buildcuda
+      (lib.withFeature true "ucc")
+      (lib.withFeature true "ucx")
+
+      (lib.enableFeature (!cudaSupport) "mca-dso")
+      (lib.withFeatureAs cudaSupport "cuda" cudaPackages.cuda_cudart)
+      (lib.enableFeature cudaSupport "dlopen")
+
+      (lib.withFeatureAs stdenv.isLinux "pmix" (lib.getDev pmix))
+      (lib.withFeatureAs stdenv.isLinux "pmix-libdir" "${lib.getLib pmix}/lib")
```

Comment:
Motivation: prevents autoconfigure from guessing stuff.
Didn't migrate all of the flags because out of scope. The ones migrated I had already touched for other reasons